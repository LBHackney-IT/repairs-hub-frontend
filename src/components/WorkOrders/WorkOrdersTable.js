import PropTypes from 'prop-types'
import WorkOrderRow from './WorkOrderRow'
import { Button } from '../Form'
import { Table, THead, TBody, TR, TH } from '../Layout/Table'

const WorkOrdersTable = ({
  workOrders,
  pageNumber,
  handlePageClick,
  pageSize,
}) => (
  <div>
    <h1 className="lbh-heading-h1">Manage work orders</h1>

    <Table className="govuk-!-margin-top-5 govuk-!-width-full hackney-work-order-table">
      <THead>
        <TR className="lbh-body">
          <TH scope="col" className="lbh-body-xs">
            Reference
          </TH>
          <TH scope="col" className="lbh-body-xs">
            Date raised
          </TH>
          <TH scope="col" className="lbh-body-xs">
            Priority
          </TH>
          <TH scope="col" className="lbh-body-xs">
            Property
          </TH>
          <TH scope="col" className="lbh-body-xs">
            Status
          </TH>
          <TH scope="col" className="lbh-body-xs">
            Trade
          </TH>
          <TH scope="col" className="lbh-body-xs">
            Description
          </TH>
        </TR>
      </THead>
      <TBody>
        {workOrders.map((job, index) => (
          <WorkOrderRow key={index} {...job} />
        ))}
      </TBody>
    </Table>
    <div className="page-navigation govuk-!-padding-bottom-5">
      {pageNumber > 1 && (
        <Button
          label="Previous page"
          onClick={() => handlePageClick(pageNumber - 1)}
          type="submit"
        />
      )}
      {pageNumber && workOrders.length >= pageSize && (
        <Button
          label="Next page"
          onClick={() => handlePageClick(pageNumber + 1)}
          type="submit"
          className="right-page-button"
        />
      )}
    </div>
  </div>
)

WorkOrdersTable.propTypes = {
  workOrders: PropTypes.arrayOf(
    PropTypes.shape({
      reference: PropTypes.number,
      dateRaised: PropTypes.string,
      lastUpdated: PropTypes.instanceOf(Date),
      priority: PropTypes.string,
      property: PropTypes.string,
      status: PropTypes.string,
      description: PropTypes.string,
    })
  ).isRequired,
  pageNumber: PropTypes.number,
  handlePageClick: PropTypes.func,
  pageSize: PropTypes.number.isRequired,
}

export default WorkOrdersTable
