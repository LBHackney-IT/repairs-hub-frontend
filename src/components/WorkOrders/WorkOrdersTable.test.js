import { render } from '@testing-library/react'
import UserContext from '../UserContext/UserContext'
import WorkOrdersTable from './WorkOrdersTable'

describe('WorkOrdersTable component', () => {
  const props = {
    workOrders: [
      {
        reference: 10000012,
        dateRaised: '2021-01-12T16:24:26.632Z',
        lastUpdated: new Date('2021-01-13T16:24:26.632Z'),
        priority: '2 [E] EMERGENCY',
        property: '1 Pitcairn House St Thomass Square',
        owner: '',
        status: 'In Progress',
        tradeDescription: 'Electrical - EL',
        description:
          'ALPHA- Pitcairn house op stucl behind carpark gates from power network pls remedy AND Communal: Door entry; Residents locked out/in',
      },
    ],
    pageNumber: 1,
    handlePageClick: jest.fn(),
  }

  it('should render properly', () => {
    const user = {
      name: 'A Contractor',
      email: 'a.contractor@hackney.gov.uk',
      roles: ['contractor'],
      hasRole: true,
      hasAgentPermissions: false,
      hasContractorPermissions: true,
      hasContractManagerPermissions: false,
      hasAuthorisationManagerPermissions: false,
      hasAnyPermissions: true,
    }

    const { asFragment } = render(
      <UserContext.Provider value={{ user }}>
        <WorkOrdersTable
          workOrders={props.workOrders}
          pageNumber={props.pageNumber}
          handlePageClick={props.handlePageClick}
        />
      </UserContext.Provider>
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
