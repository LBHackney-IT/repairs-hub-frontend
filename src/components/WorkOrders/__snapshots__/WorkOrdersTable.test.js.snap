// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkOrdersTable component should render properly 1`] = `
<DocumentFragment>
  <div>
    <h1
      class="lbh-heading-h1"
    >
      Manage work orders
    </h1>
    <table
      class="govuk-table lbh-table govuk-!-margin-top-5 govuk-!-width-full hackney-work-order-table"
    >
      <thead
        class="govuk-table__head"
      >
        <tr
          class="govuk-table__row lbh-body"
        >
          <th
            class="govuk-table__header lbh-body-xs"
            scope="col"
          >
            Reference
          </th>
          <th
            class="govuk-table__header lbh-body-xs"
            scope="col"
          >
            Date raised
          </th>
          <th
            class="govuk-table__header lbh-body-xs"
            scope="col"
          >
            Priority
          </th>
          <th
            class="govuk-table__header lbh-body-xs"
            scope="col"
          >
            Property
          </th>
          <th
            class="govuk-table__header lbh-body-xs"
            scope="col"
          >
            Status
          </th>
          <th
            class="govuk-table__header lbh-body-xs"
            scope="col"
          >
            Trade
          </th>
          <th
            class="govuk-table__header lbh-body-xs"
            scope="col"
          >
            Description
          </th>
        </tr>
      </thead>
      <tbody
        class="govuk-table__body"
      >
        <tr
          class="govuk-table__row govuk-table__row--clickable lbh-body-s hackney-work-order-table"
          data-ref="10000012"
        >
          <td
            class="govuk-table__cell lbh-body-xs"
          >
            <a
              class="lbh-link"
              href="/work-orders/10000012"
            >
              10000012
            </a>
          </td>
          <td
            class="govuk-table__cell lbh-body-xs"
          >
            12 Jan 2021
            <div
              class="work-order-hours"
            >
              16:24
            </div>
          </td>
          <td
            class="govuk-table__cell lbh-body-xs"
          >
            2 [E] EMERGENCY
          </td>
          <td
            class="govuk-table__cell lbh-body-xs"
          >
            1 Pitcairn House St Thomass Square
          </td>
          <td
            class="govuk-table__cell"
          >
            <span
              class="govuk-tag lbh-tag lbh-tag--status-in-progress lbh-body-xs work-order-status"
            >
              In progress
            </span>
          </td>
          <td
            class="govuk-table__cell lbh-body-xs"
          >
            Electrical - EL
          </td>
          <td
            class="govuk-table__cell description lbh-body-xs"
          >
            <p
              class="truncate-description truncate-line-5"
            >
              ALPHA- Pitcairn house op stucl behind carpark gates from power network pls remedy AND Communal: Door entry; Residents locked out/in
            </p>
          </td>
        </tr>
      </tbody>
    </table>
    <div
      class="page-navigation govuk-!-padding-bottom-5"
    />
  </div>
</DocumentFragment>
`;
